corehq/apps/builds/jadjar.py

```diff
@@ -132,7 +134,7 @@ def pack(self, files, jad_properties=None):
         buffer = StringIO(self.jar)
         with ZipFile(buffer, 'a', ZIP_DEFLATED) as zipper:
             for path in files:
-                zipper.writestr(path, files[path])
+                zipper.writestr(path, files[path].replace(BASE_ADDRESS, J2ME_ADDRESS))
```

Comment:
This seems a little dangerous. Is it ever possible that there could be more than one occurrence of `BASE_ADDRESS` within `files[path]`? If so, then this is not safe.

A safer approach (assuming the occurrence of `BASE_ADDRESS` that you're replacing is always at the beginning of the path) would be this:

```py
rel_path = files[path][:len(BASE_ADDRESS)].lstrip("/")  # assume always POSIX paths
new_path = os.path.join(J2ME_ADDRESS, rel_path)
```